packages/shared/src/introspectMachine.ts

```diff
@@ -351,7 +351,17 @@ function collectEnterables(ctx: TraversalContext, node: XState.StateNode) {
     return;
   }
 
-  const enterableNodes = [node, ...node.initialStateNodes];
+  const enterableNodes = new Set([node]);
+  const initialLeafNodes = node.initialStateNodes;
```

Comment:
You can confirm how it behaves here: https://codesandbox.io/s/small-glitter-strroo?file=/src/index.js